### DISCUSSION TITLE
How would one go about forcing a particular language for certain routes?

### DISCUSSION THREAD

**Discussion by jonschmidt** [amannn/next-intl#1138] [Q&A]
We have 2 different apps on our site, a marketing site in next-js, and a logged in site built in angular. The marketing site uses next-auth to determine the language. The dashboard stores the users language in their user settings (in our database).

We are starting to use more logged in functionality into the next app, and need to determine how to have consistency, so logged in users always see the language they have set in their user settings.

My initial idea is to store the locale on the jwt authentication token and check that in our middleware, then force the locale there using rewrites, but I have not thought this through too much. Anybody dealt with something similar or have any thoughts?

**Comment by amannn** [comment]
The [App Router mixed routing example](https://next-intl-docs.vercel.app/examples#app-router-mixed-routing) ([demo](https://x.com/jamannnnnn/status/1801283932501037298)) seems somewhat related to your use case, maybe that can provide some inspiration.

I guess the bigger question for me would be if you want i18n routing on the marketing site and/or the logged in app. If you don't need it for the logged in app, you can already solve providing a locale in `i18n.ts` in that case similar to the example mentioned above by reading the JWT from a header. As for the marketing site, my question would be if you can rely on a JWT being present, I guess this would only be set once a user is logged in? In the linked example the locale is saved in a cookie that is shared among the public site and the internal app, ensuring that both are in sync, but not requiring an active user session.
